lib/middleware.js

```diff
@@ -1,10 +1,65 @@
 'use strict';
 
+require('dotenv').load();
+const jwt = require('jwt-simple');
+
 module.exports = function(req, res, cb) {
-  /* To be replaced with token validation and/or authentication */
-  const validated = Object.assign(req, {
-    user: 'test-user',
-    isAdmin: false
-  });
-  return cb(null, validated);
+  const whiteListPath = '/auth/';
+
+  // don't check for tokens if user is hitting an auth url
+  if (req.path.startsWith(whiteListPath)) {
+    return cb();
+  }
+
+  const token = getToken(req);
+  const payload = token ? getPayload(token) : null;
+
+  if (payload && payload.id) {
+    req.user = payload;
+    if (process.env.trustedUsers) {
+      const trustedUsers = JSON.parse(process.env.trustedUsers);
+      req.isTrustedUser = trustedUsers.indexOf(payload.id) !== -1;
+    } else {
+      req.isTrustedUser = true; // if no env var, all OSM users are trusted
+    }
+  } else {
+    req.user = null;
```

Comment:
Do we need to set `req.user`? It sounds like in this case, `req.isTrustedUser` is set to false, which means it triggers a 401 status code, and the request doesn't proceed.

Reply:
I don't think we need to set `res.isTrustedUser` but adding `req.user` is probably useful so that end-points can use that information to fill in a `user` field or so where applicable. I'll change things around to remove `req.isTrustedUser`.